mlperf_logging/result_summarizer/result_summarizer.py

```diff
@@ -282,21 +302,30 @@ def summarize_results(folder, ruleset):
             allowed_benchmarks = _ALLOWED_BENCHMARKS_V07
         elif ruleset == '1.0.0':
             allowed_benchmarks = _ALLOWED_BENCHMARKS_V10
+        csv_header.extend(allowed_benchmarks)
         for benchmark in allowed_benchmarks:
             if benchmark in benchmark_scores:
                 row += '{:.2f},'.format(benchmark_scores[benchmark])
             else:
                 row += ','
 
         # Construct postfix portion of the row.
+        csv_header.append('details_url')
         row += '{},'.format(_details_url(desc, ruleset))
+        csv_header.append('code_url')
         row += '{},'.format(_code_url(desc, ruleset))
 
         rows[_row_key(desc)] = row
 
     # Print rows in order of the sorted keys.
     for key in sorted(rows):
         print(rows[key])
+        # Add rows to csv object
+        if csv_obj is not None:
+            # Add the header above the first results row
+            if csv_obj.csv_rows == list():
+                csv_obj.add_row(csv_header)
+            csv_obj.add_row([*csv.reader([rows[key]])][0])
```

Comment:
This way of "setting `row` to be a comma-separated string first, then parsing back into list using `csv.reader`, just to be converted into csv string again" seems unnecessary to me. I would:
- Either have `row` to be a list to begin with, then passing the file handler or `sys.stdout` to the `csv.writer`'s constructor depending on the value of the `--csv` flag.
- Or not using `csv.writer` at all and write `row` to file directly.

Reply:
Thank you, I chose the latter